functions/Restore-DbaBackup.ps1

```diff
@@ -0,0 +1,215 @@
+function Restore-DbaBackup
+{
+<#
+.SYNOPSIS 
+Restores a SQL Server Database from a set of backupfiles
+
+.DESCRIPTION
+Scans a given folder for Full, Differential and Log backups. 
+OR
+Takes a set of folder arrays and processes them for Full, Differential and Log backups.
+
+These are then filtered and restored to a specified SQL Server intance and file location
+
+The backup LSN chain and RecoveryForkID will also be checked to ensure the restore is valid
+
+It can also generate restore scripts, both as part of a restore or as it's only action
+
+.PARAMETER SqlServer
+The SQL Server instance. 
+
+.PARAMETER SqlCredential
+Allows you to login to servers using SQL Logins as opposed to Windows Auth/Integrated/Trusted. 
+
+.PARAMETER Path
+Path to SQL Server backup file. This can be a full, differential or log backup file.
+
+.PARAMETER Files
+A Files object(s) containing SQL Server backup files. 
+
+.PARAMETER RestoreLocation
+Path to restore the SQL Server backups to on the target inance
```

Comment:
Had a look, and to do the data/log stuff I'd need to push a change up for Read-DbaBackupHeader as it doesn't return file type atm.

Happy to do that if it's not too much overhead for yourselves?